RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -256,19 +256,21 @@ class AuthenticationService: NSObject {
             }
         }
         
-        let client = clientType.init(homeServer: homeserverURL, unrecognizedCertificateHandler: { [weak self] certificate in
-            guard let self = self else { return false }
-            
-            var isTrusted = false
-            let semaphore = DispatchSemaphore(value: 0)
-            
-            self.delegate?.authenticationService(self, needsPromptFor: certificate) { didTrust in
-                isTrusted = didTrust
-                semaphore.signal()
-            }
-            
-            semaphore.wait()
-            return isTrusted
+        let client = clientType.init(homeServer: homeserverURL, unrecognizedCertificateHandler: { /*[weak self] certificate*/ _ in
+            // Tchap: Don't allow unrecognized certificates !
+            return false
```

Comment:
I would prefer to force nil the handler here instead of implement a fake one

Reply:
Fixed in last commit.